src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java

```diff
@@ -56,6 +56,8 @@
 public final class NonBlockingStatsDClient implements StatsDClient {
 
     private static final int PACKET_SIZE_BYTES = 1400;
+    private static final int SOCKET_TIMEOUT_MS = 100;
+    private static final int SOCKET_BUFFER_BYTES = -1;
```

Comment:
Could you confirm where this default comes from? Would this be the same as not providing a default and falling back on whatever the system has set?

Reply:
This would keep the exact settings what we use today:
- 100ms timeout
- default socket write buffer size (because any size <= 0 is ignored)

Reply:
Ok great, the timeout made sense, but wasn't aware of the socket size being ignored if less than zero. Thanks!

Reply:
It's not in the library but in our own code :)

```
                if (bufferSize > 0) {
                    clientChannel.setOption(UnixSocketOptions.SO_SNDBUF, bufferSize);
                }
```